std/experimental/logger/filelogger.d

```diff
@@ -21,17 +21,36 @@ class FileLogger : Logger
       file can not be opened for writting an exception will be thrown.
       lv = The $(D LogLevel) for the $(D FileLogger). By default the
       $(D LogLevel) for $(D FileLogger) is $(D LogLevel.all).
+      createFileNameFolder = if true and fn contains a folder name, this
+      folder will be created.
 
     Example:
     -------------
     auto l1 = new FileLogger("logFile");
     auto l2 = new FileLogger("logFile", LogLevel.fatal);
+    auto l3 = new FileLogger("logFile", LogLevel.fatal, true);
     -------------
     */
     this(in string fn, const LogLevel lv = LogLevel.all) @safe
     {
+        this(fn, lv, false);
+    }
+
+    this(in string fn, const LogLevel lv, bool createFileNameFolder) @safe
+    {
+        import std.file : exists, mkdirRecurse;
+        import std.path : dirName;
+
         super(lv);
+
         this.filename = fn;
+        auto d = dirName(this.filename);
+        if (createFileNameFolder && !exists(d))
+        {
+            () @trusted { mkdirRecurse(d); }();
```

Comment:
Looking at mkdirRecurse, I think the trusted escape is OK. However, the best solution is to actually make mkdirRecurse `@safe`, and put the trusted escape on the C call inside there. If you want to do that, it would benefit others as well.